CODE_REVIEW.md

```diff
@@ -0,0 +1,50 @@
+# Are there any problems or code smells in the app?
+
+- __Resolved__: `book-search.component.html` =>We should replace the format date function as a native date pipe because the format date function is called when, change detection cycle is triggered, whereas a native date pipe is triggered when input data is modified.
+
+- `book-search.component.html` => Responsive design is not handled for list of books.
+- __Resolved__: `book-search.component.ts` => Memory leak: Unsubscribtion is  missing for subscription done on getAllBooks. Using async pipe instead of subscribing will allow you to access to data in the template without creating extra property. Angular automatically unsubscribes the observable on component destroy. This prevents the problem of orphan subscriptions and memory leaks. Also, we don't need to hold the subscription with extra property in the component.When we use async pipe here , an unsubscription is done when the component is destroyed.
+
+- __Resolved__: `reading-list.reducer.spec.ts` => Test cases are failing
+
+- __Resolved__: `books.reducer.ts` => Unused store property: "searchTerm" is updated from the reducer but is not referred anywhere in component/selector.
+
+- __Resolved__: `reading-list.reducer.ts` => failedAddToReadingList & failedRemoveFromReadingList actions are not caught in the reducer. We need to handle failure of add/remove book API calls so we can revert the state for added/removed book else it will result in inaccurate data being displayed to the user.
+
+- __Resolved__: `book-search.component.html, reading-list.component.html` => Better naming is required for loop variables in the HTML templates. Proper naming of the variables provides more readability to the code.
+
+# Are there other improvements you would make to the app? What are they and why?
+
+- There is no spinner while fetching books. It is good to show a spinner while fetching books, as network constraints might be there.- The pagination is not implemented, so only 10 books are fetched, currently matching the search criteria. Adding pagination would allow users to move through pages of book items and see more than 10 books. As an option for pagination, we can also add an infinite loading feature where users can scroll as per need to see more books.
+
+- Provide an informational message to the user for, if books are not found for the specific search term.
+
+- __Resolved__: `books.effects.ts, reading-list.effects.ts` => Constant is not used for API endpoints.
+
+- __Resolved__: `book-search.component.html` => Rather than  interpolating coverUrl `<img src="{{ b.coverUrl }}" />`, we can use property binding on src attribute `<img [src]="b.coverUrl" />`. It seems  technically better.
```

Comment:
Give proper reason while we can use property binding over interpolation. Also remove `It seems  technically better.` as it doesn't seem proper here.